### PR TITLE
decouple [Context] from the backend

### PR DESCRIPTION
**Pull request by cwong-ocaml**
`Context` feels like a front-end module along with `findlib`, so this is more preliminary work to remove direct dependencies on it from more backend modules (particularly `build_system` and `action_exec`).

Signed-off-by: Cameron Wong <cwong@janestreet.com>

### CODE REVIEW

File: src/dune/build_context.ml [hunk 1]
```diff
@@ -0,0 +1,13 @@
+open! Stdune
+open! Import
+
+type t =
+  { name : Context_name.t
+  ; build_dir : Path.Build.t
+  ; env : Env.t
+  ; ocaml : Action.Prog.t
```
Comment (rgrinberg) [inline src/dune/build_context.ml:8]: Why does ocaml need to be passed? I don't notice it being used.

Comment (cwong-ocaml) [reply src/dune/build_context.ml:8]: I could have sworn that something needed it, but on a rebuild it seems not. I've made the change.

Comment (rgrinberg) [reply src/dune/build_context.ml]: It seemed a bit suspicious because the build rules shouldn't know anything about OCaml rules.

Comment (cwong-ocaml) [reply src/dune/build_context.ml]: Agreed; I think it was leftover mistake from a previous pass I made that I overlooked.

**Comment by cwong-ocaml** [comment]
I think this is otherwise ready to be merged and relatively non-offensive/unlikely to have major objections, but I'm not particularly enthused about clicking the button myself on Friday night like this. If there are no objections (and no major merge conflicts by then), I'll do it on Monday.

**Comment by cwong-ocaml** [comment]
It looks like the travis failure is related to something about `kill` not being found. Is this a CI problem?

**Comment by aalekseyev** [comment]
I pushed to master a change that increases the coq compilation timeout and also ignores this failure of "kill" command. Hopefully that makes travis happier. 

Anyway, I think we can merge this because it looks like a simple refactoring with no potential for behavior change.

**Pull request merged**